vimiv/gui/thumbnail.py

```diff
@@ -143,6 +144,75 @@ def n_rows(self) -> int:
         """Return the number of rows."""
         return math.ceil(self.count() / self.n_columns())
 
+    def _first_index(self) -> int:
+        """Return the index of the first thumbnail to be rendered."""
+        result: int
+        if api.settings.thumbnail.load_behind.value == -1:
+            result = 0
+        else:
+            result = max(
+                0, self.current_index() - api.settings.thumbnail.load_behind.value
+            )
+        return result
+
+    def _last_index(self) -> int:
+        """Return the index of the last thumbnail to be rendered."""
+        result: int
+        if len(self._paths) == 0:
+            result = 0
+        elif api.settings.thumbnail.load_ahead.value == -1:
+            result = len(self._paths) - 1
+        else:
+            result = min(
+                len(self._paths) - 1,
+                self.current_index() + api.settings.thumbnail.load_ahead.value,
+            )
+        return result
+
+    def _index_range(self) -> Tuple[int, int]:
+        """Return the first and last indexes in a tuple fitting a Python range."""
+        return (self._first_index(), self._last_index() + 1)
+
+    def _prune_index(self, index) -> None:
+        """Unload the icon associated with a particular path index."""
+        item = self.item(index)
+        # Otherwise it has been deleted in the meanwhile
+        if item is not None and (
+            api.settings.thumbnail.unload_threshold.value == -1
+            or len(self._rendered_paths) > api.settings.thumbnail.unload_threshold.value
+        ):
+            item.setIcon(ThumbnailItem.default_icon())
+            self._rendered_paths.discard(self._paths[index])
+
+    def _prune_icons_behind(self) -> None:
+        """Prune indexes before the current index."""
+        for index in range(0, self._first_index()):
+            self._prune_index(index)
+
+    def _prune_icons_ahead(self) -> None:
+        """Prune indexes after the current index."""
+        for index in reversed(range(self._last_index() + 1, len(self._paths))):
```

Comment:
Out of curiosity, why the `reversed`?

Reply:
The `reversed` is there, so that when pruning icons ahead of the current selection, it's done from the end first. If it went in normal order, there would be a gap from the icons removed after the selection. I hope I explained that reasonably well.